internal/events/subscription.go

```diff
@@ -140,13 +140,35 @@ func (s *subscription) processNewEvents() {
 				EventName:     ccEvent.EventName,
 				Payload:       ccEvent.Payload,
 			}
+			if s.ep.stream.spec.Timestamps {
+				s.getEventTimestamp(event)
+			}
 			if err := s.ep.processEventEntry(s.info, event); err != nil {
 				log.Errorf("Failed to process event: %s", err)
 			}
 		}
 	}
 }
 
+func (s *subscription) getEventTimestamp(evt *eventsapi.EventEntry) {
+	// the key in the cache is the block number represented as a string
+	blockNumber := strconv.FormatUint(evt.BlockNumber, 10)
+	if ts, ok := s.ep.stream.blockTimestampCache.Get(blockNumber); ok {
+		// we found the timestamp for the block in our local cache, assert it's type and return, no need to query the chain
+		evt.Timestamp = ts.(int64)
```

Comment:
Should we check the result of this casting?

Reply:
I think this is ok because the only code that adds to this cache is in this same function